crates/cli/src/config.rs

```diff
@@ -790,86 +852,96 @@ impl Config {
             .find(|path| path.exists())
     }
 
-    fn load_raw(config_dir: PathBuf, is_project: bool) -> RawConfig {
-        // If a config file overload has been specified, use that instead
-        if !is_project {
-            if let Some(config_path) = std::env::var_os("SPACETIME_CONFIG_FILE") {
-                return Self::load_from_file(config_path.as_ref())
-                    .inspect_err(|e| eprintln!("SPACETIME_CONFIG_FILE does not point to a valid config file: {e:#}"))
-                    .unwrap_or_default();
-            }
-        }
-        if !config_dir.exists() {
-            fs::create_dir_all(&config_dir).unwrap();
+    fn system_config_path() -> PathBuf {
+        if let Some(config_path) = std::env::var_os("SPACETIME_CONFIG_FILE") {
+            config_path.into()
+        } else {
+            let mut config_path = dirs::home_dir().unwrap();
+            config_path.push(HOME_CONFIG_DIR);
+            Self::find_config_path(&config_path).unwrap_or_else(|| config_path.join(CONFIG_FILENAME))
         }
-
-        let Some(config_path) = Self::find_config_path(&config_dir) else {
-            return if is_project {
-                // Return an empty config without creating a file.
-                RawConfig::default()
-            } else {
-                // Return a default config with http://127.0.0.1:3000 as the default server.
-                // Do not (yet) create a file.
-                // The config file will be created later by `Config::save` if necessary.
-                RawConfig::new_with_localhost()
-            };
-        };
-
-        Self::load_from_file(&config_path)
-            .inspect_err(|e| eprintln!("config file is invalid: {e:#}"))
-            .unwrap_or_default()
     }
 
     fn load_from_file(config_path: &Path) -> anyhow::Result<RawConfig> {
         let text = fs::read_to_string(config_path)?;
         Ok(toml::from_str(&text)?)
     }
 
-    pub fn load() -> Self {
-        let home_dir = dirs::home_dir().unwrap();
-        let home_config = Self::load_raw(home_dir.join(HOME_CONFIG_DIR), false);
-
+    fn load_proj_config() -> RawConfig {
         // TODO(cloutiertyler): For now we're checking for a spacetime.toml file
         // in the current directory. Eventually this should really be that we
         // search parent directories above the current directory to find
         // spacetime.toml files like a .gitignore file
         let cur_dir = std::env::current_dir().expect("No current working directory!");
-        let cur_config = Self::load_raw(cur_dir, true);
+        if let Some(config_path) = Self::find_config_path(&cur_dir) {
+            Self::load_from_file(&config_path)
+                .inspect_err(|e| eprintln!("config file {config_path:?} is invalid: {e:#}"))
+                .unwrap_or_default()
```

Comment:
I know it was like this before, but it might be worth raising the error here. I would prefer a CLI tool crashes rather than continuing with unknown defaults when there's a config error.

Reply:
I agree with you, but I'd like @jdetter to weigh in before I change any semantics more than is required for the ticket.